bucket/adb.json

```diff
@@ -1,17 +1,20 @@
 {
-    "version": "28.0.2",
+    "version": "28.0.3",
     "homepage": "https://developer.android.com/studio/releases/platform-tools.html",
     "description": "Android SDK platform-tools includes tools that interface with the Android platform, such as adb, fastboot, and systrace",
-    "url": "https://dl.google.com/android/repository/platform-tools_r28.0.2-windows.zip",
-    "hash": "6a721560633babebc74f9330d7184a23d74d18b835cc769f81bbf977575b3800",
+    "url": "https://dl.google.com/android/repository/platform-tools_r28.0.3-windows.zip",
+    "hash": "0fdbc4bcbfcc2b9437f36d7c17cf7bb318532221de54c7e71fc980f05f51bcbf",
     "bin": [
         "platform-tools\\adb.exe",
         "platform-tools\\dmtracedump.exe",
         "platform-tools\\etc1tool.exe",
         "platform-tools\\fastboot.exe",
         "platform-tools\\hprof-conv.exe"
     ],
-    "checkver": "<h4.*>([\\d.]+) \\(.*\\)</h4>",
+    "checkver": {
+        "url": "https://dl.google.com/android/repository/repository2-1.xml",
+        "re": "platform-tools_r(?<version>[\\d.]+)-windows.zip"
```

Comment:
We can use XPath to query the version and hash.

Reply:
Woo, that's a nice feature, but wiki isn't update yet.

I'll try, wait a second.

Reply:
After some testing, I think XPath make things complicated.

Google puts version code in sperated nodes:

```xml
	<remotePackage path="platform-tools">
		<!--Generated from:ab bid:5475833 branch:aosp-sdk-release-->
		<type-details xsi:type="generic:genericDetailsType"/>
		<revision>
			<major>28</major>
			<minor>0</minor>
			<micro>3</micro>
		</revision>
```

To get full version string, we have to use `concat` function, the xpath looks terrible: 

```JavaScript
// Tested with Chrome
$x("concat(//remotePackage[@path='platform-tools']/revision/major, '.', //remotePackage[@path='platform-tools']/revision/minor, '.', //remotePackage[@path='platform-tools']/revision/micro)")
// Result: "28.0.3"
```

And the worst thing is, powershell did not supported `concat` function. So regular expression is the simplest way for this.